signer/src/crypto/secret_uri.rs

```diff
@@ -115,10 +117,10 @@ impl std::str::FromStr for SecretUri {
 }
 
 /// This is returned if `FromStr` cannot parse a string into a `SecretUri`.
-#[derive(Debug, Copy, Clone, PartialEq, thiserror::Error)]
+#[derive(Debug, Copy, Clone, PartialEq, Display)]
 pub enum SecretUriError {
     /// Parsing the secret URI from a string failed; wrong format.
-    #[error("Invalid secret phrase format")]
+    #[display(fmt = "Invalid secret phrase format")]
```

Comment:
I agree with Niklas in the subxt-core PR; let's add:

```rust
#[cfg(feature = "std")]
impl std::error::Error for SecretUriError {}
```

To this and any other Error type. This means that on "std" the errors will impl `Error`